api/types/types_drivers_executor.go

```diff
@@ -132,6 +158,7 @@ type ProvidesStorageExecutorCLI interface {
 // the driver implementations of the StorageExecutor interface.
 type StorageExecutorCLI interface {
 	StorageExecutorFunctions
+	StorageExecutorWithSupported
```

Comment:
By the way, your comment had it totally right:

> Why put `StorageExecutorWithSupported` here instead of putting `StorageExecutorFunctions` in `StorageExecutorWithSupported`? This seems to modify an existing interface, such that all executors will have to be updated to be usable. Is that true? If it is true and that is your intent, read no further...

> The other way around would seeming make `StorageExecutorWithSupported` a new interface that you could check if an executor implemented before calling the `Supported` method -- thereby allowing old and new versions. I may be completely off here and speaking total nonsense.

Golang refactoring used to *suck*. There is a stdlib [tool](https://godoc.org/golang.org/x/tools/refactor/rename) for it, but the last time I used it, well after the introduction of Go dependency vendoring, there were still an outpouring of grief that the tool would fail if vendored dependendencies were used. 

And since many Go tools that integrate with Atom use static analysis of the compiled packages, if the packages fail to compile you don't get things like intellisense, etc. Not to mention that after 10 errors Go just stops telling you the build errors.

So until the refactoring for Golang improves (I need to investigate to see if it has), I tend to make additive changes via new interfaces to prevent build failures.